text/metrics/0108-naming-conventions.md

```diff
@@ -0,0 +1,23 @@
+# Metric instrument naming conventions
+
+## Purpose
+
+Names and labels for metric instruments are primarily how humans interact with metric data -- users rely on these names to build dashboards and perform analysis. The names and hierarchical structure need to be understandable and discoverable during routine exploration -- and this becomes critical during incidents.
+
+To ensure these goals and consistency in future metric naming standards, this outlines a meta-standard for these names.
+
+## Guidelines
+
+Metric names and labels exist within a single universe and a single hierarchy. Metric names and labels MUST be considered within the universe of all existing metric names. When defining new metric names and labels, consider the prior art of existing standard metrics and metrics from frameworks/libraries.
+
+Associated metrics SHOULD be nested together in a hierarchy based on their usage. Define a top-level hierarchy for common metric categories: for OS metrics, like CPU and network; for app runtimes, like GC internals. Libraries and frameworks should nest their metrics into a hierarchy as well. This aids in discovery and adhoc comparison. This allows a user to find similar metrics given a certain metric.
+
+The hierarchical structure of metrics defines the namespacing. Supporting OpenTelemetry artifacts define the metric structures and hierarchies for some categories of metrics, and these can assist decisions when creating future metrics.
+
+Common labels SHOULD be consistently named. This aids in discoverability and disambiguates similar labels to metric names.
```

Comment:
I am not sure how to interpret this w.r.t. service name. For example, [Prometheus][] guidelines specifically state

> For metrics specific to an application, the prefix is usually the application name itself.

In contrast, at Uber we structure all metrics to have a set of common infra labels (like zone/datacenter) that includes `service` name. So if have two services A and B and they both emit a metric with name "blah", then because it comes from different services there's no guarantee that the semantics of such metric are identical.

[Prometheus]: https://prometheus.io/docs/practices/naming/

Reply:
(I assume this comment relates to line 15.) 

The Prometheus guidelines, which we are following here, say that same-named metrics should be meaningful to aggregate (maybe not sensible, just meaningful). If you have a metric named "requests", maybe, with a service label, it's meaningful to aggregate those across services. So if "blah" is used across services, the different appearances **ought** to mean the same thing.

You didn't say whether you think the way Uber does this is good or not? At Lightstep we did in fact use the application prefix on our "requests" metric, and most of the time I'm glad I don't have to filter by service to build a service-specific dashboard. But once in a while, I want a generic dashboard and I can't just group by service, so it's not always clear what is best.

Reply:
I'm not sure if there's something actionable about this line.  
As it is written, it makes sense to me.  We're saying that, across broad categories of metrics, labels that mean the same thing (that is: common labels) should be named the same thing.  

This seems like enough detail for this guideline.  More detail than this belongs in semantic conventions, in my opinion.